weepickle-implicits/src/main/scala-3/com/rallyhealth/weepickle/v1/implicits/CaseClassTo.scala

```diff
@@ -0,0 +1,197 @@
+package com.rallyhealth.weepickle.v1.implicits
+
+import compiletime.{summonInline}
+import deriving.Mirror
+import scala.util.control.NonFatal
+import com.rallyhealth.weepickle.v1.core.{Abort, Annotator, ObjVisitor, NoOpVisitor, Types, Visitor}
+
+trait CaseClassToPiece extends MacrosCommon :
+  this: Types with Annotator =>
+
+  // must duplicate ToString definition here to avoid circular reference with DefaultTos
+  implicit val stringVisitor: To[String] = new SimpleTo[String] {
+    override def expectedMsg = "expected string"
+    override def visitString(cs: CharSequence): String = cs.toString
+  }
+
+  class CaseClassTo[T](
+    fromProduct: Product => T,
+    // parallel arrays in field definition order
+    fieldNames: Array[String],
+    defaultValues: Array[Option[() => AnyRef]],
+    createVisitors: => Array[Visitor[_, _]]
+  ) extends CaseR[T] :
+
+    private lazy val visitors = createVisitors
+    private val outOfBounds = -1
```

Comment:
```suggestion
    private inline val outOfBounds = -1
```

Reply:
oooo... why?

Reply:
It's used as a constant. As written, it creates a 4-byte field on every instance. [Depending on alignment](https://shipilev.net/jvm/anatomy-quarks/24-object-alignment/) JVM may use more heap. Admittedly, there won't be nearly as many of these instances as something like `ObjVisitor`, but free savings are free savings.

Reply:
done